File: opspilot-enterprise/packages/shared-types/src/case-archive.ts
export type CaseStatus = "open" | "archived" | "reviewing";
export type CaseCategory = "performance" | "availability" | "capacity" | "change_failure" | "security" | "network" | "other";

export interface CaseArchive {
  id: string;
  title: string;
  summary: string;
  category: CaseCategory;
  status: CaseStatus;
  severity: string;
  tags: string[];
  incident_refs: string[];
  root_cause_summary: string;
  resolution_summary: string;
  lessons_learned: string;
  author: string;
  created_at: string;
  archived_at: string | null;
  similarity_score: number | null;
  hit_count: number;
  knowledge_refs: string[];
}

File: opspilot-enterprise/packages/shared-schema/src/opspilot_schema/case_archive.py
from __future__ import annotations
from typing import Optional
from pydantic import BaseModel


class CaseArchive(BaseModel):
    id: str
    title: str
    summary: str
    category: str
    status: str
    severity: str
    tags: list[str] = []
    incident_refs: list[str] = []
    root_cause_summary: str
    resolution_summary: str
    lessons_learned: str
    author: str
    created_at: str
    archived_at: Optional[str] = None
    similarity_score: Optional[float] = None
    hit_count: int = 0
    knowledge_refs: list[str] = []
